src/docs/services_and_contributions.md

```diff
@@ -4,42 +4,84 @@ title: Services and Contributions
-the used `FileSystem` concretion is just a proxy sending JSON-RPC messages to
-the backend, so it needs a particular configuration and treatment. The
-navigator doesn't need to care as it will get injected a fully working
-`FileSystem` instance.
+Theia uses the DI framework InversifyJS to wire up the different components including services and contribution points.
+DI decouples the consumers of services -- i.e. the dependencies of those consumers -- from the actual creation and retrieval of those services. As an example, if you want to use a service, you neither have to instantiate it, nor do you need to manually retrieve it from somewhere. Instead, the DI container injects the services on creation of your component. The DI container resolves the dependency for you and, if necessary, even instantiates it on the fly. With that, the consumer of services doesn’t need to worry where they come from and you can easily exchange the actual implementations of services later on without having to change the consumers. The DI container works based on some configuration you provide on startup through so-called container modules.
 
-Moreover, this decoupling of construction and use, allows extensions to provide
-their very specific implementations of e.g. a `FileSystem` if needed. Still
-without touching any users of the `FileSystem` interface.
+We will provide examples on how to use dependency injection in the sections “Services” and “Contributing to contribution points” below.
 
 DI is an integral part of Theia. Therefore, we highly recommend learning at
-least the basics of [Inversify.js](http://inversify.io/).
+least the basics of [InversifyJS](http://inversify.io/). For more details, please also refer to [this article on how dependency injection works in Theia](https://eclipsesource.com/blogs/2018/11/28/how-to-inversify-in-eclipse-theia/)
+
+## Using Services
+
+To use a service in Theia, you can get it injected as a dependency using DI. Dependencies are usually specified via the interface of the service you want to retrieve. This way, you even avoid a dependency on any specific implementation, the caller just knows the interface. This allows the component providing the implementation to seamlessly replace the implementation of a service. You can even override the existing default implementation of a service without breaking any service consumers.
 
-## Services
+Please note that InversifyJS technically uses Symbols to identify interfaces, because “Interface” is not a value in JavaScript. Therefore, service interfaces usually define a Symbol with exactly the same name as the interface itself.
+
+Services, or more generically, dependencies can be injected as fields, in the constructor or in initialization functions (see following code example).
+
+```typescript
+//Injection in the constructor
+constructor(@inject(MessageService) private readonly messageService: MessageService) { }
```

Comment:
Again, I think MessageService is not a good example: for one thing, it does not implement the symbol/interface/implementation pattern.

Reply:
StorageService maybe?

Reply:
Ah, is that why you find MessageSerice hard to get? Should we maybe fix this rather in code than in the docu then and add an interface to MessageService?

Reply:
The advantage of MessageService for me is that the service itself is trivial to understand, we have docu for it and it is part of most examples

Reply:
Arghhh...I was thinking of `MessagingService` 🤦 Quite a different beast.